GoogleForCreators/web-stories-wp
# Media: Ensure user feedback when inserting unsupported file types via WordPress media modal

## Context



## Summary

This ticket become a little larger, when I looked around a different uses of the media picker in editor. In short, the link ( elements) , poster (document), publisher logo ( document ), caption ( video ) and poster ( video ), did have any feedback / error handling if an incorrect file type was selected. Now a snack message is displayed in all these different cases with an error message that matches what you are doing. 

## Relevant Technical Choices

- New PHP changes are required, so that a list of image types can be displayed. 
- Link icons, now only support valid image types and ignore SVGS. 
- Captions are now limited to VTT files and show an error message when an invalid one is selected.

## To-do

- Add some tests
- Name variable and param names better. 

## User-facing changes



## Testing Instructions



### QA






This PR can be tested by following these steps:

Test 1: Insert in compatibility file types for media dialog. 

1. Ensure that transcoding feature flag is disabled.
2. Open the editor. 
3. Click upload button. 
4. Upload an incompatible file type, such as a .mov or .pdf file. 
5. Wait for upload to complete.
6. Click insert into page. 
7. See showbar message.

Test 2: Select poster / publish logo

0. Open the editor. 
0. Click to open documents tab. 
0. Click on edit icon for poster or publisher logo
0. Upload an incompatible file type, such as a .mov or .pdf file. 
0. Wait for upload to complete.
0. Click select poster / publisher logo button
0. See showbar message.

Test 3: Select poster for video element

0. Open the editor. 
0. Insert video into canvas ( either by uploading a new one or using an existing one, not important for this test ).
0. Click on edit icon for poster
0. Upload an incompatible file type, such as a .mov or .pdf file. 
0. Wait for upload to complete.
0. Click select poster
0. See showbar message.

Test 4: Select caption file for video element

0. Open the editor. 
0. Insert video into canvas ( either by uploading a new one or using an existing one, not important for this test ).
0. Click on upload captions button
0. Upload an incompatible file type, such as a .mov or .pdf file. 
0. Wait for upload to complete.
0. Click select poster
0. See showbar message.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #6756
